packages/contract/src/util.ts

```diff
@@ -64,17 +63,8 @@ export const getContractId = (
  * Ensures that a string is hexlified.
  *
  * @param value - The value to be hexlified.
- * @param isKnownHex - Required if using hex values that need to be converted
+ *
  * @returns The input value hexlified with prefix.
  */
-export const hexlifyWithPrefix = (value: string, isKnownHex = false) => {
-  if (value.startsWith('0x')) {
-    return hexlify(value);
-  }
-
-  if (isKnownHex) {
-    return hexlify(`0x${value}`);
-  }
-
-  throw new FuelError(FuelError.CODES.UNEXPECTED_HEX_VALUE, `Value should be hex string ${value}.`);
-};
+export const hexlifyWithPrefix = (value: string) =>
```

Comment:
IMO this function is pretty redundant. Hexlifying a hex value will produce the same value. Defo an edge case and unlike but we risk this passing with `0xNotAHexVal`. May as well just call `hexlify`.

Reply:
Reverted in 23995888b8e82a863f03e60f4fd8896b7a2ab711.

As storage slots have key, values without the `0x` prefix, it [fails](https://github.com/FuelLabs/fuels-ts/actions/runs/8542109255/job/23403007707?pr=2002#step:3:1011).